### ISSUE TITLE
Beta version tasks

### ISSUE THREAD

**Issue by assafmo** [scrtlabs/secret.js#9]
A list of tasks we need to complete before we're out of beta.

🟥 - TODO  
🟨 - Implemented  
🟩 - Implemented + Tested

- 🟥 Local HD wallet with Secret Network key derivation path
  - 🟨 Amino signer
  - 🟥 Protobuf signer
- 🟥 Test Keplr
- 🟥 Test Keplr+Ledger
- 🟥 Test subscribe/ws
- 🟨 Queries
  - 🟨 getTx(txhash)
  - 🟨 txsQuery(queryString)
  - 🟩 auth
  - 🟨 authz
  - 🟨 bank
  - 🟩 compute
  - 🟨 distribution
  - 🟨 evidence
  - 🟨 feegrant
  - 🟨 gov
  - 🟨 ibc_channel
  - 🟨 ibc_client
  - 🟨 ibc_connection
  - 🟨 ibc_transfer
  - 🟨 mint
  - 🟨 params
  - 🟨 registration
  - 🟨 slashing
  - 🟨 staking
  - 🟨 tendermint
  - 🟨 upgrade
- 🟥 Transactions
  - 🟥 authz
    - 🟥 MsgExec
    - 🟨 MsgRevoke
    - 🟨 MsgGrant
  - 🟩 bank
    - 🟩 MsgSend
    - 🟩 MsgMultiSend
  - 🟩 compute
    - 🟩 MsgStoreCode
    - 🟩 MsgInstantiateContract
    - 🟩 MsgExecuteContract
  - 🟥 crisis
    - 🟥 MsgVerifyInvariant
  - 🟩 distribution
    - 🟩 MsgWithdrawDelegatorReward
    - 🟩 MsgWithdrawValidatorCommission
    - 🟩 MsgFundCommunityPool
    - 🟩 MsgSetWithdrawAddress
  - 🟥 evidence
    - 🟥 MsgSubmitEvidence
  - 🟥 feegrant
    - 🟥 MsgRevokeAllowance
    - 🟥 MsgGrantAllowance
  - 🟨 gov
    - 🟨 MsgSubmitProposal
      - 🟩 TextProposalContent
      - 🟩 CommunityPoolSpendProposalContent
      - 🟩 ParameterChangeProposalContent
      - 🟨 ClientUpdateProposalContent
      - 🟨 UpgradeProposalContent
      - 🟨 SoftwareUpgradeProposalContent - signature mismatch bug
      - 🟩 CancelSoftwareUpgradeProposalContent
    - 🟩 MsgVote
    - 🟩 MsgVoteWeighted
    - 🟩 MsgDeposit
  - 🟥 ibc_channel (low priority)
    - 🟥 MsgChannelCloseConfirm
    - 🟥 MsgRecvPacket
    - 🟥 MsgTimeout
    - 🟥 MsgTimeoutOnClose
    - 🟥 MsgChannelOpenInit
    - 🟥 MsgAcknowledgement
    - 🟥 MsgChannelOpenTry
    - 🟥 MsgChannelOpenAck
    - 🟥 MsgChannelOpenConfirm
    - 🟥 MsgChannelCloseInit
  - 🟥 ibc_client (low priority)
    - 🟥 MsgUpgradeClient
    - 🟥 MsgSubmitMisbehaviour
    - 🟥 MsgCreateClient
    - 🟥 MsgUpdateClient
  - 🟥 ibc_connection (low priority)
    - 🟥 MsgConnectionOpenTry
    - 🟥 MsgConnectionOpenAck
    - 🟥 MsgConnectionOpenConfirm
    - 🟥 MsgConnectionOpenInit
  - 🟥 ibc_transfer
    - 🟥 MsgTransfer
  - 🟩 slashing
    - 🟩 MsgUnjail
  - 🟩 staking
    - 🟩 MsgCreateValidator
    - 🟩 MsgEditValidator
    - 🟩 MsgDelegate
    - 🟩 MsgBeginRedelegate
    - 🟩 MsgUndelegate

**Comment by assafmo** [comment]
Moved to https://github.com/scrtlabs/secret.js/projects/1
